src/diffusers/models/modeling_utils.py

```diff
@@ -801,20 +821,50 @@ def _get_model_file(
-                revision=revision,
-            )
+            if revision in DEPRECATED_REVISION_ARGS and version.parse(
+                version.parse(__version__).base_version
+            ) >= version.parse("0.15.0"):
+                variant = _add_variant(weights_name, revision)
+
+                try:
+                    model_file = hf_hub_download(
+                        pretrained_model_name_or_path,
+                        filename=weights_name,
+                        cache_dir=cache_dir,
+                        force_download=force_download,
+                        proxies=proxies,
+                        resume_download=resume_download,
+                        local_files_only=local_files_only,
+                        use_auth_token=use_auth_token,
+                        user_agent=user_agent,
+                        subfolder=subfolder,
+                        revision=revision,
+                    )
+                    warnings.warn(
+                        f"You are loading the variant {variant} from {pretrained_model_name_or_path} via `revision='{variant}'` even though you can load it via `variant=`{variant}`. Loading model variants via `revision='{variant}'` is deprecated and will be removed in diffusers v1. Please use `variant='{variant}'` instead. For more information, please have a look at: ",
+                        FutureWarning,
+                    )
+                except:  # noqa: E722
+                    warnings.warn(
+                        f"You are loading the variant {variant} from {pretrained_model_name_or_path} via `revision='{variant}'`. This behavior is deprecated and will be removed in diffusers v1. One should use `variant='{variant}'` instead. However, it appears that {pretrained_model_name_or_path} currently does not have a {_add_variant(weights_name)} file in the 'main' branch of {pretrained_model_name_or_path}. \n The Diffusers team and community would be very grateful if you could open an issue: https://github.com/huggingface/diffusers/issues/new with the title '{pretrained_model_name_or_path} is missing {_add_variant(weights_name)}' so that the correct variant file can be added.",
+                        FutureWarning,
+                    )
+                    model_file = None
+            else:
+                # Load from URL or cache if already cached
+                model_file = hf_hub_download(
+                    pretrained_model_name_or_path,
+                    filename=weights_name,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    proxies=proxies,
+                    resume_download=resume_download,
+                    local_files_only=local_files_only,
+                    use_auth_token=use_auth_token,
+                    user_agent=user_agent,
+                    subfolder=subfolder,
+                    revision=revision,
+                )
             return model_file
```

Comment:
I'm a bit confused by the logic here. I think the second error message about "main" not having a file seems to indicate we're checking the main branch but we still use the passed in revision to check a branch? Maybe I'm missing something

Reply:
Yes, so essentially we should deprecate the behavior of doing `revision="fp16"`. Now we might have cases where this works because the branch still exists, but we do want to tell the user that this behavior is deprecated, so we throw a warning.

Now there are two possibilities:
- 1.) There is already a "fp16" variant file on the main branch -> in this case the user can be directly guided to the new usage
- 2.) There is not yet a "fp16" variant file. In this case we probably should do it manually, therefore I'm trying to have the user open an issue

Reply:
You're 100% right @williamberman - the code was bad here. Refactored it, should be better now I hope